main.go

```diff
@@ -1,21 +1,132 @@
+	mongoPassword := os.Getenv("MONGO_PASSWORD")
+	mongoAddress := os.Getenv("MONGO_ADDRESS")
+	mongoPort := os.Getenv("MONGO_PORT")
+
+	// Set up the MongoDB client with SCRAM-SHA-1 authentication
+	clientOptions := options.Client().ApplyURI("mongodb://" + mongoAddress + ":" + mongoPort).
+		SetAuth(options.Credential{
+			Username:   mongoUsername,
+			Password:   mongoPassword,
+			AuthMechanism: "SCRAM-SHA-256",
+		})
+
+	client, err = mongo.Connect(context.TODO(), clientOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Check the connection
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Println("Connected to MongoDB!")
+}
 
 func handleRequests() {
-	//http.HandleFunc("/", homePage)
-	log.Fatal(http.ListenAndServe(":8000", nil))
+	router := mux.NewRouter()
+
+	// // Example handle request
+	// router.HandleFunc("/", homePage).Methods(http.MethodGet)
+
+	// Users CRUD Operations
+	router.HandleFunc("/users", func(w http.ResponseWriter, r *http.Request) {
+		users.CreateUser(w, r, client.Database("schedule_db").Collection("users"))
+	}).Methods(http.MethodPost)
+
+	router.HandleFunc("/users", func(w http.ResponseWriter, r *http.Request) {
+		users.GetUsers(w, r, client.Database("schedule_db").Collection("users"))
+	}).Methods(http.MethodGet)
+
+	router.HandleFunc("/users/{username}", func(w http.ResponseWriter, r *http.Request) {
+		users.GetUser(w, r, client.Database("schedule_db").Collection("users"))
+	}).Methods(http.MethodGet)
+
+	router.HandleFunc("/users/{username}", func(w http.ResponseWriter, r *http.Request) {
+		users.UpdateUser(w, r, client.Database("schedule_db").Collection("users"))
+	}).Methods(http.MethodPut)
+
+	router.HandleFunc("/users/{username}", func(w http.ResponseWriter, r *http.Request) {
+		users.DeleteUser(w, r, client.Database("schedule_db").Collection("users"))
+	}).Methods(http.MethodDelete)
+
+	// Schedules CRUD Operations
+	router.HandleFunc("/schedules", func(w http.ResponseWriter, r *http.Request) {
+		schedules.CreateSchedule(w, r, client.Database("schedule_db").Collection("schedules"))
+	}).Methods(http.MethodPost)
+
+	router.HandleFunc("/schedules", func(w http.ResponseWriter, r *http.Request) {
+		schedules.GetSchedules(w, r, client.Database("schedule_db").Collection("schedules"))
+	}).Methods(http.MethodGet)
+
+	router.HandleFunc("/schedules/{schedule}", func(w http.ResponseWriter, r *http.Request) {
+		schedules.GetSchedule(w, r, client.Database("schedule_db").Collection("schedules"))
+	}).Methods(http.MethodGet)
+
+	router.HandleFunc("/schedules/{schedule}", func(w http.ResponseWriter, r *http.Request) {
+		schedules.UpdateSchedule(w, r, client.Database("schedule_db").Collection("schedules"))
+	}).Methods(http.MethodPut)
+
+	router.HandleFunc("/schedules/{schedule}", func(w http.ResponseWriter, r *http.Request) {
+		schedules.DeleteSchedule(w, r, client.Database("schedule_db").Collection("schedules"))
+	}).Methods(http.MethodDelete)
```

Comment:
More of a stylistic opinion for readability, would we want to move each grouping of endpoints into a sub handle method to reduce the lines of code in the handleRequests(). ie: handleUserRequests(), handleScheduleRequests(). A good rule of thumb I have gone by is if you need to scroll then the method is starting to get too big. Non blocking tho, we can do this in a later ticket.

Reply:
I think this would be a good ticket for the next sprint. Going forward we should also update the README with style guides for code and PRs. Try to standardize a few things.